lmdeploy/vl/model/deepseek_vl2.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import os
+from contextlib import redirect_stdout
+from typing import Dict, List
+
+import torch
+from transformers import AutoConfig
+
+from lmdeploy.utils import get_logger
+from lmdeploy.vl.model.base import VISION_MODELS, VisonModel
+
+logger = get_logger('lmdeploy')
+
+
+def check_deepseek_vl2_install():
+    """check deepseek_vl install."""
+    try:
+        import deepseek_vl2  # noqa: F401
+    except ImportError:
+        raise ImportError('To use DeepSeek-VL2, please install deepseek_vl2 by '
+                          '`pip install git+https://github.com/deepseek-ai/DeepSeek-VL2.git'
+                          ' --no-deps`')
+
+
+@VISION_MODELS.register_module()
+class DeepSeek2VisionModel(VisonModel):
+    """DeepSeek2 vision model."""
+
+    _arch = 'DeepseekV2ForCausalLM'
+
+    @classmethod
+    def match(cls, config: AutoConfig):
+        """check whether the config match the model."""
+        if hasattr(config, 'language_config') and hasattr(config, 'vision_config'):
+            arch = config.language_config.get('architectures', [None])[0]
+            return arch == cls._arch
+        return False
+
+    def build_preprocessor(self):
+        check_deepseek_vl2_install()
+        from deepseek_vl2.models.processing_deepseek_vl_v2 import DeepseekVLV2Processor
```

Comment:
may fail with transformers>=4.48.0. maybe we should include this in PR description ? 
related issue https://github.com/deepseek-ai/DeepSeek-VL2/issues/45